packages/calypso-e2e/src/rest-api-client.ts

```diff
@@ -450,6 +450,34 @@ export class RestAPIClient {
 		return response;
 	}
 
+	/**
+	 * Updates the user's settings.
+	 *
+	 * @param {SettingsParams} details Key/value attributes to be set for the user.
+	 * @returns { { [key: string]: string | number } } Generic object.
```

Comment:
I opted to not define a specific type here because a POST call to the `/me/settings` endpoint simply parrots back whatever that was passed in as parameter.

Reply:
**Suggestion (non-blocking)**: if you want to tie that response type to be the exact same as what is passed in, you can actually do this with generic types! Not required, but might be helpful.

So for example:
```typescript
async setMySettings<T extends SettingsParams>( details: T ): Promise< T > {}
```
(You can also use a more helpful name than T)

The return type will then strictly match what is passed in! 

Again, 100% optional, but wanted to suggest in case it was helpful! 🙂